### PR TITLE
Improve Cypress unstable tests

### PR DESCRIPTION
**Pull request by ndricimrr**
**Description**
Finding elements within iframes introduces flakyness in Cypress tests as iframe is sometimes not loaded in time. 

Changes proposed in this pull request:
- Add more robust iframe retrieving methods to better pick up iframes. Following guide in: https://www.cypress.io/blog/2020/02/12/working-with-iframes-in-cypress/ 

**Related issue(s)**

Tests failing randomly

### CODE REVIEW

File: test/e2e-test-application/e2e/support/commands.js [hunk 1]
```diff
@@ -157,6 +157,63 @@ Cypress.Commands.add('getIframeBody', (getIframeOpts = {}, index = 0, containerS
...
+      // wraps "body" DOM element to allow
+      // chaining more Cypress commands, like ".find(...)"
+      // https://on.cypress.io/wrap
+      .then(cy.wrap)
+  );
+});
+
+// only works if iframe and parent window are of the same origin
+Cypress.Commands.add('getIframeDocumentSameOrigin', () => {
+  // get the iframe > document
```
Comment (ndricimrr) [inline test/e2e-test-application/e2e/support/commands.js:179]:
```suggestion
  // get the iframe > document 
```

File: test/e2e-test-application/e2e/support/commands.js [hunk 2]
```diff
@@ -157,6 +157,63 @@ Cypress.Commands.add('getIframeBody', (getIframeOpts = {}, index = 0, containerS
     .iframe();
 });
 
+// More robust iframe retrival methods based on: https://www.cypress.io/blog/2020/02/12/working-with-iframes-in-cypress/
+// retrieves the <iframe /> element directly
```
Comment (ndricimrr) [inline test/e2e-test-application/e2e/support/commands.js:161]:
```suggestion
// retrieves the <iframe /> element directly 
```

File: test/e2e-test-application/e2e/support/commands.js [hunk 3]
```diff
@@ -157,6 +157,63 @@ Cypress.Commands.add('getIframeBody', (getIframeOpts = {}, index = 0, containerS
...
+      .should('not.be.undefined')
+      // wraps "body" DOM element to allow
+      // chaining more Cypress commands, like ".find(...)"
+      // https://on.cypress.io/wrap
+      .then(cy.wrap)
+  );
+});
+
+Cypress.Commands.add('getIframeWindow', () => {
+  // get the iframe > contentwindow
```
Comment (ndricimrr) [inline test/e2e-test-application/e2e/support/commands.js:194]:
```suggestion
  // get the iframe > contentwindow 
```

**Comment by viktorsperling** [comment]
#### 2 Objects:

1. Step `Encoded ViewURL Search Params with Decorators` of `fiddle-theming.spec.js` has errors for me locally. Is the URL in the test assertion correct? `http://localhost:8080/examples/microfrontends/customUserSettingsMf.html?someURL=http%3A%2F%2Fsome.url%2Ffoo%2Fbar&sap-theme=green`

2. The new commands are only used for the fiddle part of the e2e tests but I see some applications for the angular-1 tests, too. Is it intended that these commands are only used in fiddle?

**Comment by ndricimrr** [comment]
> #### 2 Objects:
> 1. Step `Encoded ViewURL Search Params with Decorators` of `fiddle-theming.spec.js` has errors for me locally. Is the URL in the test assertion correct? `http://localhost:8080/examples/microfrontends/customUserSettingsMf.html?someURL=http%3A%2F%2Fsome.url%2Ffoo%2Fbar&sap-theme=green`
> 2. The new commands are only used for the fiddle part of the e2e tests but I see some applications for the angular-1 tests, too. Is it intended that these commands are only used in fiddle?

**1. Step:** 🤔  the URL should be correct. Is Fiddle running properly for you when you test. What errors are you getting. Let me know

**2. Step**: The new commands are added as [Cypress Commands ](https://docs.cypress.io/api/cypress-api/custom-commands#Usage). When added in this way they can be used all over the e2e cypress tests, be it fiddle or angular.

**Review by viktorsperling — APPROVED** [review]
LGTM;

**Pull request merged**